api/v1alpha1/destination_types.go

```diff
@@ -28,9 +28,5 @@ type S3 struct {
 	// +optional
 	Bucket string `json:"bucket,omitempty"`
 	// +optional
-	AccessKeyID string `json:"accessKeyID,omitempty"`
```

Comment:
I think we misunderstood each other. Those fields should still be used, but their value can be an environment variable, which will later be substituted, e.g.:
```
# ...
    accessKeyID: $MY_SECRET_ENV
env:
- name: MY_SECRET_ENV
# ...
```

Reply:
Ah I see, at which point in the code will this substitution be processed or is it some kind of build in K8s?